### PR TITLE
set hha_hash to equal instance_prefix

### PR DESCRIPTION
**Pull request by JettTech**
### Updates:
 - sets `hha_hash` to equal the `intance_prefix`, as they represent the same value and the real value is needed for chaperone.DEVELOP
 - includes the update to ensure the ws is open before proceeding to step 4 (ready) - https://github.com/Holo-Host/chaperone/issues/61

### CODE REVIEW

**Comment by JettTech** [comment]
Note: This includes the update to resolve the RPC error that @robbie is working on.

File: src/index.ts [hunk 1]
```diff
@@ -556,7 +556,7 @@ class Chaperone {
 	}
 
 	if ( this.mode === Chaperone.DEVELOP || this.mode === Chaperone.HCC ) {
-	    this.hha_hash		= "QmV1NgkXFwromLvyAmASN7MbgLtgUaEYkozHPGUxcHAbSL"; // some random digest hash
+	    this.hha_hash		= this.opts.instance_prefix;
 	    log("Skipping Resolver Hostname lookup and using fake HHA ID: %s", this.hha_hash );
```
Comment (mjbrisebois) [inline src/index.ts:560]:
```suggestion
	    log("Skipping Resolver Hostname lookup and using configured instance prefix as HHA ID: %s", this.hha_hash );
```

File: src/index.ts [hunk 2]
```diff
@@ -762,7 +762,7 @@ class Chaperone {
 			curr_state	= 2;
 		    if ( this.opened === true )	// setting up wormhole
 			curr_state	= 3;
-		    if ( this.anonymous === true || this.wormhole_ready === true ) // ready
+		    if ( this.opened === true  || (this.anonymous === true || this.wormhole_ready === true) ) // ready
```
Comment (mjbrisebois) [inline src/index.ts:765]: This change will make state 4 always overwrite state 3.  Did you mean?
```suggestion
		    if ( this.opened === true  && (this.anonymous === true || this.wormhole_ready === true) ) // ready
```

**Review by mjbrisebois — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (JettTech) [reply src/index.ts:765]: :sweat_smile: Yes.

**Review by mjbrisebois — APPROVED** [review]
Tagging issue https://github.com/Holo-Host/chaperone/issues/61 which this PR fixes

LGTM.  Though it is always nice to include tests that prove we don't make this mistake again.

**Pull request merged**